### PR TITLE
Adding refresh relationships to smart action

### PR DESCRIPTION
**Pull request by Tolsa**
- adding a new action for a clean usecase on demo
- Implementing logic + refresh cache new feature

### CODE REVIEW

File: lib/forest_liana/collections/customer.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@ class Forest::Customer
 
   collection :Customer
 
+  action 'Boom test'
```
Comment (louisteenbrink) [inline lib/forest_liana/collections/customer.rb]: Boom test ?

**Review by louisteenbrink — DISMISSED** [review]

File: lib/forest_liana/collections/customer.rb (hunk 1 above)
Comment (Tolsa) [reply lib/forest_liana/collections/customer.rb]: my bad I was interrupted by Lunchr and did a quick test but forgot to remove it, will update this accordingly

**Review by louisteenbrink — APPROVED** [review]

File: Gemfile [hunk 2]
```diff
@@ -5,6 +5,7 @@ git_source(:github) do |repo_name|
   "https://github.com/#{repo_name}.git"
 end
 
+
```
Comment (SeyZ) [inline Gemfile:8]: To be removed

File: Gemfile.lock [hunk 3]
```diff
@@ -247,6 +247,7 @@ DEPENDENCIES
   listen (>= 3.0.5, < 3.2)
   mixpanel_client
   pg
+  pry
```
Comment (SeyZ) [inline Gemfile.lock:250]: Why?

File: lib/forest_liana/collections/company.rb [hunk 4]
```diff
@@ -7,6 +7,15 @@ class Forest::Company
 
   action 'Mark as Live'
 
+  action 'New emitted transaction', fields: [{
+    field: "Beneficiary company",
+    description: "name of the company who will receive the transaction",
```
Comment (SeyZ) [inline lib/forest_liana/collections/company.rb:12]: The description should starts with a capital letter and ends with a `.`

File: app/controllers/forest/companies_controller.rb [hunk 5]
```diff
@@ -43,4 +43,32 @@ def upload_legal_docs
     # Once the upload is finished, send a success message to the admin user in the UI.
     render json: { success: 'Legal documents are successfully uploaded.' }
   end
+
+  def new_emitted_transaction
+
```
Comment (SeyZ) [inline app/controllers/forest/companies_controller.rb:48]: Useless space

File: app/controllers/forest/companies_controller.rb [hunk 6]
```diff
@@ -43,4 +43,32 @@ def upload_legal_docs
     # Once the upload is finished, send a success message to the admin user in the UI.
     render json: { success: 'Legal documents are successfully uploaded.' }
   end
+
+  def new_emitted_transaction
+    
+    # here is the logic behind the smart action form
```
Comment (SeyZ) [inline app/controllers/forest/companies_controller.rb:49]: Useless comment

File: app/controllers/forest/companies_controller.rb [hunk 7]
```diff
@@ -43,4 +43,32 @@ def upload_legal_docs
...
+      vat_amount: Faker::Number.number(4),
+      fee_amount: Faker::Number.number(4),
+      status: ['to_validate', 'validated', 'rejected'].sample,
+      note: Faker::Lorem.paragraph,
+      amount: amount,
+      emitter_bic: Faker::Code.nric,
+      beneficiary_bic: Faker::Code.nric
+    )
+
+    #success message and refresh cache
```
Comment (SeyZ) [inline app/controllers/forest/companies_controller.rb:68]: The comment is not explicit and this is the main purpose of this example. You should explain in details what you're doing here in the `refresh` option

**Review by louisteenbrink — APPROVED** [review]

**Comment by SeyZ** [comment]
@louisteenbrink @Tolsa I think we are currently not intransigent enough when reviewing the code :-)

**Review by SeyZ — CHANGES_REQUESTED** [review]

File: app/controllers/forest/companies_controller.rb (hunk 7 above)
Comment (arnaudbesnier) [reply app/controllers/forest/companies_controller.rb:68]: And "refresh cache" is not the right wording, as there is no cache (same thing for the PR title).

File: lib/forest_liana/collections/company.rb [hunk 8]
```diff
@@ -7,6 +7,15 @@ class Forest::Company
 
   action 'Mark as Live'
 
+  action 'New emitted transaction', fields: [{
```
Comment (arnaudbesnier) [inline lib/forest_liana/collections/company.rb:10]: Shouldn't actions name start with a verb to be consistent with our current convention?

File: app/controllers/forest/companies_controller.rb [hunk 9]
```diff
@@ -43,4 +43,32 @@ def upload_legal_docs
     # Once the upload is finished, send a success message to the admin user in the UI.
     render json: { success: 'Legal documents are successfully uploaded.' }
   end
+
+  def new_emitted_transaction
+    
+    # here is the logic behind the smart action form
+    attrs = params.dig('data','attributes', 'values')
+    beneficiary_company_id = attrs['Beneficiary company']
+    emitter_company_id = params.dig('data','attributes')["ids"][0]
```
Comment (arnaudbesnier) [inline app/controllers/forest/companies_controller.rb:52]: "ids" => 'ids' to keep consistent

**Pull request closed without merging**